core/src/test/java/org/testcontainers/utility/MountableFileTest.java

```diff
@@ -0,0 +1,77 @@
+package org.testcontainers.utility;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.assertFalse;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertTrue;
+
+public class MountableFileTest {
+
+    @Test
+    public void forClasspathResource() throws Exception {
+        final MountableFile mountableFile = MountableFile.forClasspathResource("mappable-resource/test-resource.txt");
+
+        performChecks(mountableFile);
+    }
+
+    @Test
+    public void forClasspathResourceWithAbsolutePath() throws Exception {
+        final MountableFile mountableFile = MountableFile.forClasspathResource("/mappable-resource/test-resource.txt");
+
+        performChecks(mountableFile);
+    }
+
+    @Test
+    public void forClasspathResourceFromJar() throws Exception {
+        final MountableFile mountableFile = MountableFile.forClasspathResource("docker-java.properties");
```

Comment:
Fair point. I wanted to make it be something that's specific and ~ only present in one/few JARs so that we can be sure that the full classpath is searched. i.e. it felt like searching for a resource that's trivially findable in any JAR might not be all that thorough. I used `docker-java.properties` but this is admittedly a fairly lazy choice 🙉 

I guess the real non-lazy solution would be to create a synthetic JAR with some interesting resources in it for testing purposes (maybe published with jitpack or as a local `system` scoped dependency). WDYT?